gprofiler/profilers/perf.py

```diff
@@ -77,11 +81,21 @@ def _get_perf_cmd(self) -> List[str]:
             str(self._mmap_sizes[self._type]),
         ] + self._extra_args
 
+    def check_if_needs_restart(self) -> None:
+        """Checks if perf used memory exceeds threshold, and if it does, restarts perf"""
+        if (
+            time.monotonic() - self._start_time >= self._restart_after_s
+            and Process(self._process.pid).memory_info().rss >= self._perf_memory_usage_threshold  # type: ignore
```

Comment:
What's the `type: ignore` for? psutil?

Reply:
`gprofiler/profilers/perf.py:88: error: Item "None" of "Optional[Popen[Any]]" has no attribute "pid"`
same issue as in line 134. self._process is initialized with None value

Reply:
Okay. So in such cases - this function expects `self._process` to be non-None, and using `type: ignore` weakens the static linting.

Instead, what is preferable, is to provide additional help to the static linter so that it's able to do its job. In this case, you can do that by adding `assert self._process is not None` at the beginning of the function - it's indeed a bug if the function is called while this field is not initialized.

The only times it's far to use `type: ignore` is when a package doesn't have type hints, or particular methods/classes etc don't have type hints. You DON'T use `type: ignore` to solve implicit cases like here (something is `None` but we don't expect it to be `None`, ...)